src/regex.rs

```diff
@@ -139,3 +160,26 @@ fn is_trivial_regex(s: &regex_syntax::Expr) -> Option<&'static str> {
         _ => None,
     }
 }
+
+struct RegexVisitor<'v, 't: 'v> {
+    cx: &'v LateContext<'v, 't>,
+    spans: HashSet<Span>,
+}
+
+impl<'v, 't: 'v> Visitor<'v> for RegexVisitor<'v, 't> {
+    fn visit_expr(&mut self, expr: &'v Expr) {
+        if let Some(span) = is_expn_of(self.cx, expr.span, "regex") {
```

Comment:
I see two avenues of optimization:

* [ ] check for Blocks instead of Exprs
* [ ] only consider Blocks whose expr's type matches regex::Regex